packages/@lwc/template-compiler/src/shared/types.ts

```diff
@@ -229,9 +239,23 @@ export interface ForOf extends DirectiveParentNode {
     iterator: Identifier;
 }
 
+export interface ScopedSlotContent extends DirectiveParentNode {
+    type: 'ScopedSlotContent';
+    directive: SlotDataDirective;
```

Comment:
_ScopedSlotFragment aka ScopedSlotContent(previously)_

So we(Nolan & I) debated about this. Originally the interface was like this 
```
export interface ScopedSlotFragment extends DirectiveParentNode {
    type: 'ScopedSlotFragment';
    value: Identifier;
    slotName: Literal;
}
```
We decided to add the `directive` instead for the following reasons:
1. `lwc:slot-data` is in this intersection between the template directives(for:each, if:true etc) that are for compile time(i.e they are transformed into compiled code) and the `ElementDirective`s that contain info for vnodes generated at runtime. The `<template lwc:slot-data>` fragment needs both. So we had to create a ElementDirective for it.
2. For completeness of `ElementDirective`, we had to create a `SlotDataDirective`. Now we need to add this directive information on the `ScopedSlotFragment` node. 
3. That is the reason for having a `directive` field on `ScopedSlotFragment`

Reply:
Renamed the field to `slotData`.

>  what is the value of SlotDataDirective extending from Directive versus making this a Literal?
1. The directive node contains the location of the directive itself. If we want to convert `slotData` field to an object instead of a `Literal`, then it will need a `type` and `location` field, which is essentially what `SlotDataDirective` has. So instead of creating a another interface like `SlotData` why not keep it consistent and have a `SlotDataDirective`
2. `lwc:slot-data` is a directive and `SlotDataDirective` makes the enum complete.